livekit-agents/livekit/agents/pipeline/agent_output.py

```diff
@@ -187,16 +189,28 @@ async def _synthesize_task(self, handle: SynthesisHandle) -> None:
 
 
 @utils.log_exceptions(logger=logger)
-async def _str_synthesis_task(
-    tts_text: str, transcript: str, handle: SynthesisHandle
+async def _read_transcript_task(
+    transcript_source: AsyncIterable[str] | str, handle: SynthesisHandle
 ) -> None:
-    """synthesize speech from a string"""
+    if isinstance(transcript_source, str):
+        handle._tr_fwd.push_text(transcript_source)
+    else:
+        async for seg in transcript_source:
```

Comment:
last nit: we should do that inside its own task otherwise it'll wait to get the whole transcript before synthesizing speech. This may add latency

Reply:
Isn't it in its own task already? Sorry if I'm misunderstanding, I'm not super familiar with asyncio

Reply:
It is, but I guess you need to create another task. Otherwise the `tts.synthesize` method will never be called until transcript_source is done.

Reply:
(sorry for the double ping - sent a message on slack too). _read_transcript_task is called with

```
  read_transcript_atask = asyncio.create_task(
    _read_transcript_task(transcript_source, handle)
)
```
inside the tts.synthesize loop on the first frame. What would the second task do here?